security/symmetric/test/_symmetric_key_test.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+var ArgumentError = require('azure-iot-common').errors.ArgumentError;
+var SymmetricKeySecurityClient  = require('../lib/symmetric_key').SymmetricKeySecurityClient ;
+var sinon = require('sinon');
+var assert = require('chai').assert;
+var crypto = require("crypto");
+var equals = require('array-equal');
+
+
+describe('symmetric key', function () {
+  describe('getRegistrationId', function() {
+    it('returns the registrationId', function(done) {
+      var fakeRegistrationId = 'registrationId';
+      var fakeSymmetricKey = 'fakeKey';
+      var client = new SymmetricKeySecurityClient(fakeSymmetricKey, fakeRegistrationId);
+      client.getRegistrationId(function(err, returnedId) {
```

Comment:
>err [](start = 40, length = 3)

not seeing any test where the callback path returns an error? for both methods?

Reply:
I see - it cannot fail. wondering why it's async then.

---
In reply to: [231623580](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231623580) [](ancestors = 231623580,231366832)

Reply:
Because I was trying to keep signature symmetry with the other security clients. 

---
In reply to: [231752348](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231752348) [](ancestors = 231752348,231623580,231366832)

Reply:
I see where you're coming form - this being said security clients seem to operate very differently from each other so I'm not sure in that case what symmetry buys us. 
If we were sharing the same interface with the TPM security client for example, i'd understand - but there's very little in common.

---
In reply to: [231999004](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231999004) [](ancestors = 231999004,231752348,231623580,231366832)